### ISSUE TITLE
[Bug] Already-installed app (Zoom) does not seem to register its allowed connection properly with 3.7.1 anymore?

### ISSUE THREAD

**Issue by krzemien77** [henrypp/simplewall#1495]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

3.7.1

### Windows version

Windows 10 22H2 (all latest updates etc.)

### Steps to reproduce

1. Open Zoom app (pre-installed before update to 3.7.1, latest version 5.15.5 (19404) (64-bit) )
2. Allow in 'Network connection was blocked'
3. No difference
4. Rinse and repeat - to no avail.
5. Removal from the list does not help.
6. Restart does not help either.

Screenshot enclosed.

NB I find it strange that Simplewall reports Zoom as 'unsigned'? See another screenshot - perhaps issue lies elsewhere?

Version 3.6.7 does work fine, but there's no SHA-256 feature shown in there. Also, it was possible to disable apps's signatures - it this option does not seem to be present in 3.7.1 any more?

Am reverting again to 3.6.7 in the meantime.

### Expected behavior

Filter / rule with 'allowed' connection to be added to the config

### Actual behavior

Connection remains blocked by the latest Simplewall v3.7.1

### Logs

![Capture](https://github.com/henrypp/simplewall/assets/141131408/103fd4f2-6df2-43a5-9c8e-5a86f2ac1f85)
![Capture 1](https://github.com/henrypp/simplewall/assets/141131408/afa67780-0157-424b-9e72-1d7a396976d9)

**Comment by henrypp** [comment]
even me see that zoom.exe is your list is non-existing file

**Comment by krzemien77** [comment]
What do you mean 'non-existing file'? Did you check the view from Explorer on the left - and zoom.exe highlighted in there?

Also, if it's non-existing, then where the connection prompt would be coming from?

Would you mind re-read what I wrote please - and fix?

**Comment by krzemien77** [comment]
From 3.6.7:
![Capture](https://github.com/henrypp/simplewall/assets/141131408/0a46c4d3-1900-4a4e-a5dd-07392296b888)

**Comment by henrypp** [comment]
zoom.exe is highlighted as non-existent file, in 367 i dont know because it is selected

> Also, if it's non-existing, then where the connection prompt would be coming from?

i dont know,im not telepath

**Comment by krzemien77** [comment]
Look mate, Zoom is installed on this PC, and 3.7 seems buggy and for some reasons thinks it's not.

3.6.7 is not buggy (in that respect) and recognises Zoom okay.

I supplied screenshots above to demonstrate it and that's all I had to say.

Do what you please but refrain from rude remarks and responses.

**Comment by krzemien77** [comment]
Did not see, or check anything else: literally got updated to v3.7.1 earlier today (having reverted to v3.6.7 from v3.7 two days ago or so due to the crashes) and spotted after GUI popped up that Zoom is displayed as not present.

The queries I'd be really asking here are:

1. Why its SHA-256 is empty in Simplewall (not sure how this is obtained by your app)?
2. Despite that, why v3.6.7 is fine, and v3.7.1 is not, i.e.: what's changed?

I can always re-install Zoom and see if that helps. But find it strange that v3.6.7. remember it in the rules's list, and I do not have to do anything, but v3.7.1 seems it;s not installed at all.

**Comment by krzemien77** [comment]
Please also note that Zoom is labelled as unsigned app in v3.6.7.

What does that actually mean?

**Comment by dethknite** [comment]
I would recommend uninstalling zoom and downloading the latest installation client (or at minimal downloading the latest install client) and then reinstalling it. The "not signed" means the zoom.exe file is not code-signed, which is abnormal for the actual install from Zoom. I would also remove the zoom rule and let it recreate a new one after launching.

![RvnsxBwRW9](https://github.com/henrypp/simplewall/assets/965431/41ff3234-385a-4b08-8cf3-09d884b0625c)

**Comment by krzemien77** [comment]
Yeah, might do that - re-install - but everything was fine prior to updating to the latest release.
Also, there's second instance of Zoom installed for another user, so might see if the behaviour is the same.
Currently still using v3.6.7 where all is fine.

**Comment by krzemien77** [comment]
Just after I removed the rule and re-launched Zoom prior to its possible re-installation:

![image](https://github.com/henrypp/simplewall/assets/141131408/966c0bef-bd70-43b1-9077-d667eae3ea50)

**Comment by krzemien77** [comment]
Having just removed and re-installed Zoom (latest version, downloaded at source) - still via v3.6.7:

![image](https://github.com/henrypp/simplewall/assets/141131408/d4d6413f-1ef4-47ff-adc1-ce848b82917b)

![image](https://github.com/henrypp/simplewall/assets/141131408/a25bbd0c-6b08-43a4-b684-2ef98b8bfbb5)

**Comment by krzemien77** [comment]
The situation with another user account is the same.

Via v3.6.7:

![Capture](https://github.com/henrypp/simplewall/assets/141131408/c642f219-cae5-496a-bef5-674e34b65f0f)

Once upgrade to v3.7.1 takes place:

![Capture2](https://github.com/henrypp/simplewall/assets/141131408/443b7f5e-2b3b-4d9b-94ee-95d16780206d)

![Capture3](https://github.com/henrypp/simplewall/assets/141131408/60a1b4f8-df5c-494b-abef-19352805522c)

Zoom client's version is the same in both cases.

Now, what just made me think is this: Zoom seems installing in user's profiles rather than in Program Files or similar.

Would that be the factor as effectively there are two instances of this app installed on this PC - and Simplewall might not handle that well?

**Comment by krzemien77** [comment]
Entirely out of my curiosity as admittedly I've not been lurking here for a while - since the above entry, in fact: this problem I reported above is seemingly addressed now, as I did allow simplewall to be updated to v3.7.7 some week ago or so and can confirm that all other things being equal and no changes to configuration, Zoom install etc., the app behaviour is now as expected.
So the inevitable query: did you manage to find the culprit?
